subsys/net/ip/utils.c

```diff
@@ -470,16 +469,8 @@ int z_vrfy_net_addr_pton(sa_family_t family, const char *src,
 		return -EINVAL;
 	}
 
-	memset(str, 0, sizeof(str));
-
-	nlen = z_user_string_nlen((const char *)src, sizeof(str), &err);
-	if (err) {
-		return -EINVAL;
-	}
```

Comment:
Why this check was removed? After all if the string is not a valid length, we can just return -EINVAL in that case and let the application deal with it instead of crashing.

Reply:
Done.